subprojects/logging/src/main/java/org/gradle/internal/logging/sink/OutputEventRenderer.java

```diff
@@ -117,6 +127,9 @@ public void restore(Snapshot state) {
     }
 
     public ColorMap getColourMap() {
+        if (colourMap == null) {
+            colourMap = new DefaultColorMap();
```

Comment:
:x: This is not thread safe. One thread can read `colourMap` not null, before the other thread finished initializing `DefaultColorMap`. I reckon this is called from `attachAnsiConsole`, but the method is public. So to be safe, it needs to acquire the lock.